sky/backends/cloud_vm_ray_backend.py

```diff
@@ -368,24 +365,36 @@ def add_ray_task(self,
                 for k, v in env_vars.items())
 
         logger.debug('Added Task with options: '
-                     f'{name_str}{cpu_str}{resources_str}{num_gpus_str}')
+                     f'{cpu_str}{resources_str}{num_gpus_str}')
         self._code += [
             sky_env_vars_dict_str,
             textwrap.dedent(f"""\
         script = {bash_script!r}
         if run_fn is not None:
             script = run_fn({gang_scheduling_id}, gang_scheduling_id_to_ip)
 
-        log_path = os.path.expanduser({log_path!r})
 
         if script is not None:
             sky_env_vars_dict['SKY_NUM_GPUS_PER_NODE'] = {int(math.ceil(num_gpus))!r}
             ip = gang_scheduling_id_to_ip[{gang_scheduling_id!r}]
-            sky_env_vars_dict['SKY_NODE_RANK'] = ip_rank_map[ip]
+            rank = ip_rank_map[ip]
+
+            if cluster_ips_map is not None:
+                idx_in_cluster = cluster_ips_map[ip]
+                if cluster_ips_map[ip] == 0:
+                    node_name = 'head'
+                else:
+                    node_name = f'worker{{idx_in_cluster}}'
+                name_str = f'{{node_name}}, rank={{rank}},'
+                log_path = os.path.expanduser(os.path.join({log_dir!r}, f'{{node_name}}.log'))
```

Comment:
1. How about `<node_name>-<private ip>.log`? Just to give users more info if they need to debug.
2. I think L392 should use the same format, because of the reason you mentioned (we may have 1-node tasks in an n-node cluster).